cmd/meroxa/root/apps/deploy.go

```diff
@@ -492,19 +492,13 @@ func (d *Deploy) deployApp(ctx context.Context, imageName, gitSha, specVersion s
 	return nil, nil
 }
 
-// buildApp will call any specifics to the turbine library to prepare a directory that's ready
-// to compress, and build, to then later on call the specific command to deploy depending on the language.
-func (d *Deploy) buildApp(ctx context.Context) (err error) {
-	return d.turbineCLI.Build(ctx, d.appName, true)
-}
-
 // getAppImage will check what type of build needs to perform and ultimately will return the image name to use
 // when deploying.
 func (d *Deploy) getAppImage(ctx context.Context) (string, error) {
 	d.logger.StartSpinner("\t", "Checking if application has processes...")
 	var fqImageName string
 
-	needsToBuild, err := d.turbineCLI.NeedsToBuild(ctx, d.appName)
+	needsToBuild, err := d.turbineCLI.NeedsToBuild(ctx)
```

Comment:
So I don't think we will need this anymore, since we always have to build the app to generate the space. Fortunately, we don't need to compile it, but rather just run it with the spec builder server (similar to the run server) to get the spec.

Reply:
this is still necessary for other languages. For go, it checks if it has functions, through the spec, same as ruby. This is to see if it's needed to upload source etc...

Reply:
Ah yes, true, I think I confused this with another function